File: composite/java/CompositeClient.java
import java.util.ArrayList;
import java.util.List;

class Component {
  void operation() {
    System.out.println("Component instance has been created");
  }
  void add(Component component) {}
  void remove(Component component) {}
  Component getChild(int index) {
    return null;
  }
}

class Leaf extends Component {

  @Override
  void operation() {
    System.out.println("Leaf instance has been created");
  }
}

class Composite extends Component {
  List list = new ArrayList<Component>();
  @Override
  void add(Component component) {
    list.add(component);
  }

  @Override
  void remove(Component component) {
    list.remove(component);
  }

  @Override
  Component getChild(int index) {
    return (Component) list.get(index);
  }
}

public class CompositeClient {
  public static void main(String[] args) {
    Leaf leaf = new Leaf();
    Composite composite = new Composite();
    composite.add(leaf);
    composite.getChild(0).operation();
  }
}

File: composite/python/CompositeClient.py
#!/usr/bin/env python
# coding:utf8
# @Date    : 2018/7/26
# @Author  : Koon
# @Link    : koon.cool
# When I wrote this, only God and I understood what I was doing. Now, God only knows.

"""
    ┌─┐       ┌─┐
 ┌──┘ ┴───────┘ ┴──┐
 │                 │
 │       ───       │
 │  ─┬┘       └┬─  │
 │                 │
 │       ─┴─       │
 │                 │
 └───┐         ┌───┘
     │         │
     │         │
     │         │
     │         └──────────────┐
     │                        │
     │                        ├─┐
     │                        ┌─┘
     │                        │
     └─┐  ┐  ┌───────┬──┐  ┌──┘
       │ ─┤ ─┤       │ ─┤ ─┤
       └──┴──┘       └──┴──┘
           神兽保佑
           永无BUG!
"""


class Component(object):
    def operation(self):
        pass

    def add(self, obj):
        pass

    def remove(self, obj):
        pass

    def get_child(self, index):
        pass


class Leaf(Component):
    def __repr__(self):
        return "a leaf"

    def operation(self):
        print("Leaf instance has been created")


class Composite(Component):
    def __init__(self):
        self._array = list()

    def add(self, obj):
        self._array.append(obj)

    def remove(self, obj):
        self._array.remove(obj)

    def get_child(self, index):
        return self._array[index]


if __name__ == "__main__":
    composite = Composite()
    leaf = Leaf()
    composite.add(leaf)
    composite.get_child(0).operation()
